tests/test_planning.py

```diff
@@ -54,24 +57,34 @@ def test_task_plan():
     heuristic = utils.create_task_planning_heuristic("hadd", init_atoms,
                                                      task.goal, ground_nsrts,
                                                      env.predicates, objects)
-    skeleton, _, _ = task_plan(init_atoms,
-                               task.goal,
-                               ground_nsrts,
-                               reachable_atoms,
-                               heuristic,
-                               timeout=1,
-                               seed=123)
+    task_plan_generator = task_plan(init_atoms,
+                                    task.goal,
+                                    ground_nsrts,
+                                    reachable_atoms,
+                                    heuristic,
+                                    timeout=1,
+                                    seed=123)
+    skeleton, _, metrics = next(task_plan_generator)
     assert len(skeleton) == 2
     assert isinstance(skeleton[0], _GroundNSRT)
     assert isinstance(skeleton[1], _GroundNSRT)
+    total_num_skeletons = 1
+    with pytest.raises(ApproachFailure) as e:
+        for _, _, metrics in task_plan_generator:
+            total_num_skeletons += 1
+            assert metrics["num_skeletons_optimized"] == total_num_skeletons
```

Comment:
this is testing that the metrics copying works as intended, right? just curious if you remove the `metrics.copy()` would this test break?

Reply:
it doesn't break, but i'll add another check for this